src/constraint_framework/4C_constraint_framework_submodelevaluator_nullspace.cpp

```diff
@@ -0,0 +1,178 @@
+// This file is part of 4C multiphysics licensed under the
+// GNU Lesser General Public License v3.0 or later.
+//
+// See the LICENSE.md file in the top-level for license information.
+//
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#include "4C_constraint_framework_submodelevaluator_nullspace.hpp"
+
+#include "4C_fem_condition.hpp"
+#include "4C_fem_discretization_nullspace.hpp"
+#include "4C_fem_general_elementtype.hpp"
+#include "4C_io_pstream.hpp"
+#include "4C_linalg_utils_sparse_algebra_math.hpp"
+#include "4C_linalg_utils_sparse_algebra_print.hpp"
+#include "4C_structure_new_timint_base.hpp"
+#include "4C_structure_new_timint_basedataglobalstate.hpp"
+
+FOUR_C_NAMESPACE_OPEN
+
+/*----------------------------------------------------------------------------*
+ *----------------------------------------------------------------------------*/
+Constraints::SubmodelEvaluator::NullspaceConstraintManager::NullspaceConstraintManager(
+    std::shared_ptr<Core::FE::Discretization> discret_ptr)
+{
+  discret_ptr_ = discret_ptr;
+
+  std::vector<const Core::Conditions::Condition*> KSPcond;
+  discret_ptr_->get_condition("KrylovSpaceProjection", KSPcond);
+  int numcond = KSPcond.size();
+  int numsolid = 0;
+
+  const Core::Conditions::Condition* kspcond = nullptr;
```

Comment:
Variabel names only differ in their capitalization. Can you come up with less cryptic and better distinguishable names?